packages/clay-tooltip/src/TooltipProvider.tsx

```diff
@@ -79,35 +93,44 @@ const TooltipBase: React.FunctionComponent<{
 	// Using `any` type since TS incorrectly infers setTimeout to be from NodeJS
 	const timeoutIdRef = useRef<any>();
 	const targetRef = useRef<HTMLElement | null>(null);
+	const titleNodeRef = useRef<HTMLElement | null>(null);
 	const tooltipRef = useRef<HTMLElement | null>(null);
 
 	const handleHide = ({target}: any) => {
-		const dataTitle = target && target.getAttribute('data-title');
+		const dataTitle =
+			titleNodeRef.current &&
+			titleNodeRef.current.getAttribute('data-title');
 
 		if (dataTitle) {
-			targetRef.current = null;
-
 			target.removeEventListener('click', handleHide);
-			target.setAttribute('title', dataTitle);
-			target.removeAttribute('data-title');
+			titleNodeRef.current!.setAttribute('title', dataTitle);
+			titleNodeRef.current!.removeAttribute('data-title');
```

Comment:
It would be nice to get rid of these `!` assertions (because they reduce safety). I think you can do that with a minor tweak:

```diff
diff --git a/packages/clay-tooltip/src/TooltipProvider.tsx b/packages/clay-tooltip/src/TooltipProvider.tsx
index 38da5c4c9..4765d3549 100644
--- a/packages/clay-tooltip/src/TooltipProvider.tsx
+++ b/packages/clay-tooltip/src/TooltipProvider.tsx
@@ -97,14 +97,16 @@ const TooltipBase: React.FunctionComponent<{
 	const tooltipRef = useRef<HTMLElement | null>(null);
 
 	const handleHide = ({target}: any) => {
-		const dataTitle =
-			titleNodeRef.current &&
-			titleNodeRef.current.getAttribute('data-title');
+		if (!titleNodeRef.current) {
+			return;
+		}
+
+		const dataTitle = titleNodeRef.current.getAttribute('data-title');
 
 		if (dataTitle) {
 			target.removeEventListener('click', handleHide);
-			titleNodeRef.current!.setAttribute('title', dataTitle);
-			titleNodeRef.current!.removeAttribute('data-title');
+			titleNodeRef.current.setAttribute('title', dataTitle);
+			titleNodeRef.current.removeAttribute('data-title');
 
 			titleNodeRef.current = null;
 			targetRef.current = null;
```

Reply:
Good catch. I was having issues with ts inferring that it wouldn't be possible to be null, however I can imagine that any refactoring in the future would cause this to be problematic.